smarts/core/sumo_traffic_simulation.py

```diff
@@ -107,6 +108,21 @@ def __repr__(self):
     def __str__(self):
         return repr(self)
 
+    def _destroy(self):
+        if atexit:
+            atexit.unregister(self._destroy)
+
+        if self._traci_conn:
+            self._traci_conn.close()
+
+        # Terminate sumo subprocess
+        if self._sumo_proc:
+            self._sumo_proc.stdin.close()
+            self._sumo_proc.stdout.close()
+            self._sumo_proc.stderr.close()
+            self._sumo_proc.terminate()
+            self._sumo_proc.wait()
```

Comment:
I think this has a bit of overlap with `_close_traci_and_pipes()`. It might best to combine the overlap.

Reply:
The `self._close_traci_and_pipes()` function is now reused inside the `self._destroy()` function to avoid repeated code.